be/src/env/env.cpp

```diff
@@ -0,0 +1,80 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#include "env/env.h"
+
+#include <fmt/format.h>
+
+#include <regex>
+#include <type_traits>
+
+#include "env/env_hdfs.h"
+#include "env/env_posix.h"
+#include "env/env_s3.h"
+
+namespace starrocks {
+
+class EnvRegistry {
+public:
+    using FactoryFunc = Env::FactoryFunc;
+
+    static EnvRegistry& Instance() {
```

Comment:
Forbidden the copy-constructor and assignment operator.

Reply:
updated